nmstate-varlink/varlink_service_test.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+import libnmstate
+import varlink
+import threading
+
+from nmstate_varlink import ServiceRequestHandler, service
+
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceIPv4
+from libnmstate.schema import InterfaceIPv6
+from libnmstate.schema import DNS
+from libnmstate.schema import Route
+from libnmstate.schema import RouteRule
+
+
+address = "unix:/tmp/nmstate"
+# address = "tcp:127.0.0.1:6000"
+
+
+@pytest.fixture()
+def varlink_server():
+    try:
+        server = varlink.ThreadingServer(address, ServiceRequestHandler)
+        server_thread = threading.Thread(target=server.serve_forever)
+        server_thread.daemon = True
+        server_thread.start()
+        return server
+    except OSError:
+        pass
+
+
+@pytest.fixture()
+def varlink_client():
+    client = varlink.Client(address)
```

Comment:
Why are you using a fixture for this?

Reply:
to void implementing client object separately for each test functions.

Reply:
Since it is just one line and using it does not allow to parameterize the address it seems to add unnecessary complexity to me.